core/modules/replica/HttpProcessor.cc

```diff
@@ -168,217 +430,835 @@ void HttpProcessor::_getReplicationLevel(qhttp::Request::Ptr req,
+
+                size_t numOnlineQservChunks = numOrphanQservChunks;
+                for (auto&& entry: onlineQservLevels) {
+                    maxObservedLevel = max(maxObservedLevel, entry.first);
+                    numOnlineQservChunks += entry.second;
+                }
+
+                size_t numAllQservChunks = 0;
+                for (auto&& entry: allQservLevels) {
+                    maxObservedLevel = max(maxObservedLevel, entry.first);
+                    numAllQservChunks += entry.second;
+                }
+
+                size_t numOnlineReplicationSystemChunks = numOrphanReplicationSystemChunks;
+                for (auto&& entry: onLineReplicationSystemLevels) {
+                    maxObservedLevel = max(maxObservedLevel, entry.first);
+                    numOnlineReplicationSystemChunks += entry.second;
+                }
+
+                size_t numAllReplicationSystemChunks = 0;
+                for (auto&& entry: allReplicationSystemLevels) {
+                    maxObservedLevel = max(maxObservedLevel, entry.first);
+                    numAllReplicationSystemChunks += entry.second;
+                }
+
+                // Pre-initialize the database-specific result with zeroes for all
+                // levels in the range of [0,maxObservedLevel]
+
+                json databaseJson;
+
+                for (int level = maxObservedLevel; level >= 0; --level) {
+                    databaseJson["levels"][level]["qserv"      ]["online"]["num_chunks"] = 0;
+                    databaseJson["levels"][level]["qserv"      ]["online"]["percent"   ] = 0.;
+                    databaseJson["levels"][level]["qserv"      ]["all"   ]["num_chunks"] = 0;
+                    databaseJson["levels"][level]["qserv"      ]["all"   ]["percent"   ] = 0.;
+                    databaseJson["levels"][level]["replication"]["online"]["num_chunks"] = 0;
+                    databaseJson["levels"][level]["replication"]["online"]["percent"   ] = 0.;
+                    databaseJson["levels"][level]["replication"]["all"   ]["num_chunks"] = 0;
+                    databaseJson["levels"][level]["replication"]["all"   ]["percent"   ] = 0.;
+                }
+
+                // Fill-in non-blank areas
+
+                for (auto&& entry: onlineQservLevels) {
+                    unsigned int const level = entry.first;
+                    size_t const numChunks   = entry.second;
+                    double const percent     = 100. * numChunks / numOnlineQservChunks;
+                    databaseJson["levels"][level]["qserv"]["online"]["num_chunks"] = numChunks;
+                    databaseJson["levels"][level]["qserv"]["online"]["percent"   ] = percent;
+                }
+                for (auto&& entry: allQservLevels) {
+                    unsigned int const level = entry.first;
+                    size_t const numChunks   = entry.second;
+                    double const percent     = 100. * numChunks / numAllQservChunks;
```

Comment:
we're sure there's no case (some kind of error condition or something? where `numAllQservChunks` can be 0?

Reply:
Good catch! Reinforced